src/notation/internal/notationproject.cpp

```diff
@@ -0,0 +1,236 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "notationproject.h"
+
+#include "engraving/engravingproject.h"
+#include "engraving/scoreaccess.h"
+
+#include "notationerrors.h"
+#include "masternotation.h"
+
+#include "log.h"
+
+using namespace mu::notation;
+using namespace mu::engraving;
+
+NotationProject::NotationProject()
+{
+    m_engravingProject = EngravingProject::create();
+    m_masterNotation = std::shared_ptr<MasterNotation>(new MasterNotation());
+}
+
+mu::io::path NotationProject::path() const
+{
+    return m_engravingProject->path();
+}
+
+mu::Ret NotationProject::load(const io::path& path, const io::path& stylePath, bool forceMode)
+{
+    TRACEFUNC;
+
+    std::string syffix = io::syffix(path);
+
+    //! NOTE For "mscz", "mscx" see MsczNotationReader
+    //! for others see readers in importexport module
+    auto reader = readers()->reader(syffix);
+    if (!reader) {
+        return make_ret(notation::Err::FileUnknownType, path);
+    }
+
+    return load(path, stylePath, reader, forceMode);
+}
+
+mu::Ret NotationProject::load(const io::path& path, const io::path& stylePath, const INotationReaderPtr& reader, bool forceMode)
+{
+    EngravingProjectPtr project = EngravingProject::create();
+    project->setPath(path.toQString());
+
+    INotationReader::Options options;
+    if (forceMode) {
+        options[INotationReader::OptionKey::ForceMode] = forceMode;
+    }
+
+    Ms::ScoreLoad sl;
+
+    Ms::MasterScore* score = project->masterScore();
+    Ret ret = reader->read(score, path, options);
+    if (!ret) {
+        return ret;
+    }
+
+    if (!Ms::MScore::lastError.isEmpty()) {
+        LOGE() << Ms::MScore::lastError;
+    }
+
+    project->setupMasterScore();
```

Comment:
In general, the loading process is ugly right now, I will try to make it better in the next PR